### DISCUSSION TITLE
width property for CustomMetricGroup

### DISCUSSION THREAD

**Discussion by ericxinzhang** [cdklabs/cdk-monitoring-constructs#450] [General]
My code looks as following. I can specify `height` in `CustomMonitoringProps`. I am wondering why `CustomMetricGroup` does not have a `width` property?

I understand that for high level APIs (e.g. `monitor<ResourceType>`) it's easier for the library to calculate width dynamically. However for `CustomMetricGroup` may we have a `width` property?

```Javascript
const monitoring = new MonitoringFacade(this, 'MyDashboard');

monitoring.monitorCustom({
  height: 5,
  humanReadableName: 'API',
  metricGroups: [
    {
        title: 'API Errors',
        // I wish I could specify `width` here
        graphWidgetLegend: cw.LegendPosition.RIGHT,
        metrics:[
          new cloudwatch.Metric({...}),
          new cloudwatch.Metric({...}),
       ],
    },
    {
        title: 'API Latency',
        graphWidgetLegend: cw.LegendPosition.RIGHT,
        metrics:[
          new cloudwatch.Metric({...}),
          new cloudwatch.Metric({...}),
       ],
    }
  ],
});
```

**Comment by ghost** [comment]
This would be a feature request and is sort of captured as part of #66. I don't think there's a particular reason why it doesn't exist, although we may need to be careful in how to handle overflowing content (i.e. should we error out, or just let it happen, if that's allowed?).

**Reply by ericxinzhang** [reply]
Thanks Eugene! I tried editing the source of my dashboard, setting width of one widget from 6 to 8. CloudWatch just pushes the widget to next row and right aligned. IMO it's totally acceptable, and cdk-monitoring-constructs should just let it happen.

![image](https://github.com/cdklabs/cdk-monitoring-constructs/assets/2282738/0ae1d706-1dd5-466d-8b30-c25aaa6042a2)

**Reply by ghost** [reply]
Moving to feature request: #487
